### PR TITLE
BSQuery and Pyppeteer

### PR DESCRIPTION
**Pull request by eliska-n**
TODO:

- [x] service discovery: right now, I configured to the services: `http://seacat-auth-1:8900` or `http://seacat-auth-1:3081` and it works, but I would like to see something more clever
TODO: Implement Discovery session where needed

- [x] I gave aaaaall that is needed for SCA into the bsquery descriptor - Robin told me, that pyppeteer is not the impersonator, but bs-query is, so I think it belongs there. I didn't manage to make the password secret, yet, because it must be hashed before saved into mongodb and this must be done within bs-query tech which does not exists, yet. I will do that, it is not difficult, but it is not nice. I would like to see this impersonator thing somehow simplified. Instead, it is getting, slightly, but still more complicated.

### CODE REVIEW

**Comment by eliska-n** [comment]
This will close this MR: https://github.com/TeskaLabs/asab-maestro-library/pull/35
I stole it from Elena :grimacing:

File: Site/ASAB Maestro/Descriptors/bs-query.yaml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  config:
+    "seacatauth:credentials:m2m:machine":
+      mongodb_uri: "{{MONGO_URI}}"
+      mongodb_database: auth
+  content:
+    - "mc.json": |
+        [{
+          "_id": "5f362cf58dca6ae0ad997bfc",
+          "username": "bs-query",
+          "__password": "$2b$12$mKgt17p7sSjHVQO3O00bNO3L3iRcT5MSx06.tQRY7nXL.DtA8rkaq",
```
Comment (ateska) [inline Site/ASAB Maestro/Descriptors/bs-query.yaml:45]: Passwords?

**Review by ateska — CHANGES_REQUESTED** [review]

File: Site/ASAB Maestro/Descriptors/bs-query.yaml (hunk 1 above)
Comment (eliska-n) [reply Site/ASAB Maestro/Descriptors/bs-query.yaml:45]: I'll make a tech in remote control for this - to create and store the password. I don't like it, but it is the only solution I can think of.

File: Site/ASAB Maestro/Descriptors/asab-pyppeteer.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  volumes:
+    # For a persistency of the configuration
+    - "{{SITE}}/{{INSTANCE_ID}}/conf:/conf:ro"
+
+
+asab:
+  configname: conf/asab-pyppeteer.conf
+  config:
+    seacat_auth:
+      url: http://seacat-auth-1:8900  # TODO!!!
```
Comment (eliska-n) [inline Site/ASAB Maestro/Descriptors/asab-pyppeteer.yaml:17]: TODO: Discovery session in asab-pyppeteer microservice

File: Site/ASAB Maestro/Descriptors/bs-query.yaml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    # TODO: remove after bs-query has the right folder /data 
+      path: /data/bsquery
+
+    elasticsearch:
+    # TODO: remove after bs-query is ready to read the config section
+      url: "{{ ELASTIC_HOSTS_AUTHORIZED|join(' ') }}"
+
+    auth:
+      multitenancy: "yes"
+      public_keys_url: http://seacat-auth-1:3081/openidconnect/public_keys  # TODO!!!
```
Comment (eliska-n) [inline Site/ASAB Maestro/Descriptors/bs-query.yaml:30]: TODO: Discovery session in asab

**Review by ateska — APPROVED** [review]

**Pull request merged**